### DISCUSSION TITLE
What's happening with v8.0.4 release?

### DISCUSSION THREAD

**Discussion by adrianhall** [CommunityToolkit/Datasync#153] [General]
Hey folks,

I wanted to give you all an update of what's been happening with v8.0.4 release.

First off, all the work for release has been done.  Should nothing else earth shattering come in, I'll release 8.0.4 next Friday.

There have been some good discussions around Native AOT (thanks @richard-einfinity for starting those discussions).  Given the state of Native AOT + Entity Framework Core today, I don't think v8 will support Native AOT.  I've punted those bits to the v9 release (more on that later)  However, I have made the DatasyncSerializer a public class so that you can change the JsonSerializerOptions that the client uses.  The client and server must agree on the serialization options, and its up to you to do that.

@RaoulHolzer added a PR in to support Lazy Loading Proxies in the client.  I have not tested this extensively but have added some smoke tests to ensure they don't break things.  If you have success in using Lazy Loading Proxies, please consider writing about your experience and putting a PR in for the client in-depth docs.  Explaining things that caught you out are a great way of contributing to the project.

Some infra stuff - we've switched over to centralized package management and I've updated all the packages to the last packages compatible with .NET 8, ASP.NET Core 8, and EF Core 8.

Finally, the datasync-server template is here!  So I've updated the docs to reflect that.  While I was writing docs, I also wrote about how to deal with triggers and did a few other corrections that I discovered while I was doing my blog series.

Which leads me to CommunityToolkit.Datasync v9.  Given the inevitable goodness that was announced at dotNetConf, it seems like it is a good idea to start work on CommunityToolkit.Datasync v9 so that we can target .NET 9, ASP.NET Core 9, and EF Core 9.  At this point, I can support the ASP.NET Core 9 OpenAPI support, probably Native AOT properly, and I *may* also be able to do a minimal API version of the toolkit.  (Fingers crossed - it's a big lift beyond the controllers).  What I will do is branch main into v8 (with no releases), then main (and the CI/CD pipelines) gets updated to .NET 9.

How can you contribute?

* We need more samples - particularly, I'd like to see a simple Uno and Avalonia sample (plus the docs), but I'd also like to see more complex samples including clean architecture, asp.net identity integration, and multiple linked models.  If you are working with the library, show off your work!
* We need some work on the documentation - notably, a solid "getting started" guide.  However, any improvements to the docs are good.
* If you like producing videos, feel free to do a walk through of a situation - I'm happy to link the videos into the main docs.

**Comment by richard-perry** [comment]
Thanks @adrianhall,

I'm using multiple models, azure b2c and sort of a clean architecture. I'm more or less there with a working migrator from the previous library. 

When I've got over some of my current headaches and try and work them out into the samples. Hopefully it save users a few of the stresses I've encountered along the way.

I use DI and MVVM a lot. One of the requests I was going to raise was around the latest update we've been discussing around the serializer options. Personally I'd like to see more of the configuration done upfront during registration. I was thinking of some extension methods like AddOfflineDatasync or something similar where we could pass an action to do some of the configuration. Serializer. client, auth etc. 

Cheers

Rich

**Reply by adrianhall** [reply]
That seems reasonable to me.  Why don’t you put some thought into it and put a proposal for a feature request into an issue.

**Comment by adrianhall** [comment]
Just to keep everyone aware, I was significantly disrupted by the "bomb cyclone" in the Seattle area on Tuesday.  I will be releasing 8.0.4 tomorrow, but I will not be processing any new issues / PRs until next week.
